### DISCUSSION TITLE
Some more control over the Database Notification UI

### DISCUSSION THREAD

**Discussion by myregistration** [filamentphp/filament#12745] [Ideas]
It would be great if we could have access to the current notification record in action buttons and be able to do the following: 
- hide or show markAsRead and markUnread buttons, based on current read_at value
- delete() method to remove notification record from database when action button clicked (like the X )

Be able to display an icon next to notification title or have bodyHtml()
Maybe have some control over alignment of buttons

**Comment by GeminiDev1** [comment]
For:

> delete() method to remove notification record from database when action button clicked (like the X )

The following has worked for me on V4.

```
Action::make('download')
    ->label('Download')
    ->url($downloadUrl)
    ->button()
    ->markAsRead()
    ->close()
```
